pkg/distro/packagesets/loader.go

```diff
@@ -34,15 +38,23 @@ func Load(it distro.ImageType, replacements map[string]string) rpmmd.PackageSet
 	archName := arch.Name()
 	distribution := arch.Distro()
 	distroNameVer := distribution.Name()
-	distroName := strings.SplitN(distroNameVer, "-", 2)[0]
+	// use rsplit() here, for "centos-stream-10" like distro names
+	distroName := distroNameVer[:strings.LastIndex(distroNameVer, "-")]
 	distroVersion := distribution.OsVersion()
 
-	distroSets, err := Data.Open(filepath.Join(distroName, "package_sets.yaml"))
+	// XXX: this is a short term measure, pass a set of
+	// searchPaths down the stack instead
+	var dataFS fs.FS = DataFS
+	if overrideDir := experimentalflags.String("yamldir"); overrideDir != "" {
```

Comment:
This is something that it would be really nice to log somehow so that when debugging you can clearly tell it has been overridden. I don't see any provisions for logging anywhere near this code though :/

Reply:
Great point, I added simple logging.